libs/aws/langchain_aws/agents/base.py

```diff
@@ -0,0 +1,514 @@
+        agent_summary = next((x for x in agent_summaries if x.get('agentName') == agent_name), None)
+        if agent_summary:
+            return agent_summary
+        if next_token is None:
+            return None
+
+
+class BedrockAgentFinish(AgentFinish):
+    """AgentFinish with session id information.
+
+    Parameters:
+        session_id: Session id
+    """
+
+    session_id: str
+
+    @classmethod
+    def is_lc_serializable(cls) -> bool:
+        """Check if the class is serializable by LangChain.
+
+        Returns:
+            False
+        """
+        return False
+
+
+class BedrockAgentAction(AgentAction):
+    """AgentAction with session id information.
+
+        Parameters:
+            session_id: session id
+        """
+    session_id: str
+
+    @classmethod
+    def is_lc_serializable(cls) -> bool:
+        """Check if the class is serializable by LangChain.
+
+        Returns:
+            False
+        """
+        return False
+
+
+OutputType = Union[List[BedrockAgentAction], BedrockAgentFinish]
+
+
+class BedrockAgentsRunnable(RunnableSerializable[Dict, OutputType]):
+    """
+        Invoke a Bedrock Agent
+    """
+    agent_id: Optional[str]
+    """Bedrock Agent Id"""
+    agent_alias_id: Optional[str] = _TEST_AGENT_ALIAS_ID
+    """Bedrock Agent Alias Id"""
+    client: Any
+    """Boto3 client"""
+    region_name: Optional[str] = None
+    """Region"""
+    credentials_profile_name: Optional[str] = None
+    """Credentials to use to invoke the agent"""
+    endpoint_url: Optional[str] = None
+    """Endpoint URL"""
+
+    @root_validator(skip_on_failure=True)
+    def validate_agent(cls, values: dict) -> dict:
+        if values.get("client") is not None:
+            return values
+
+        try:
+            client_params, session = cls.__get_boto_session(
+                credentials_profile_name=values["credentials_profile_name"],
+                region_name=values["region_name"],
+                endpoint_url=values["endpoint_url"]
+            )
+
+            values["client"] = session.client("bedrock-agent-runtime", **client_params)
+
+            return values
+        except ImportError:
+            raise ModuleNotFoundError(
+                "Could not import boto3 python package. "
+                "Please install it with `pip install boto3`."
+            )
+        except UnknownServiceError as e:
+            raise ModuleNotFoundError(
+                "Ensure that you have installed the latest boto3 package "
+                "that contains the API for `bedrock-runtime-agent`."
+            ) from e
+        except Exception as e:
+            raise ValueError(
+                "Could not load credentials to authenticate with AWS client. "
+                "Please check that credentials in the specified "
+                "profile name are valid."
+            ) from e
+
+    @staticmethod
+    def __get_boto_session(credentials_profile_name: str, region_name: str, endpoint_url: str):
```

Comment:
Can we move this out as a separate method.

Reply:
done